types/styled-components/index.d.ts

```diff
@@ -17,10 +18,8 @@ export type ThemedStyledProps<P, T> = P & ThemeProps<T>;
 export type StyledProps<P> = ThemedStyledProps<P, any>;
 
 export type ThemedOuterStyledProps<P, T> = P & {
+    as?: React.ComponentType | keyof JSX.IntrinsicElements;
```

Comment:
You’re right. I was thinking about this earlier. I’ll look into it when I have a chance.

Reply:
Alright so after playing around with this for a bit I don't believe this is possible in typescript. So to be honest the best way around this would be to sign a variable and give it a type which contains the required props.